### DISCUSSION TITLE
Preserve the URL in the browser

### DISCUSSION THREAD

**Discussion by darklord1807** [inertiajs/inertia#453] [Help] [answered]
Hello, 

So when i do: 
` this.$inertia
        .post(
          "http://tableloader.local/tableloader/getTables",
          {
            db: "hello"
          },
          {
            preserveScroll: true,
            replace: false,
            preserveState: true
          }
        )
        .then(() => {
          console.log(this.tables);
        });`

It changes the URL to /getTables. 

I want to keep the original link and just let it return the data and update the props. 

In Laravel I am doing: 
`       return Inertia::render('Tableloader/index', ['tables' => $tables]);`

If I use back it does not return the Prop. 

Kindly advise if this is a bug or it works as this. 

Thanks, 

Naveesh

**Comment by ajnsn** [comment, marked as answer]
@darklord1807 Inertia's Laravel adapter always returns the [$request->getRequestUri() ](https://github.com/inertiajs/inertia-laravel/blob/726b1e0d4fc399089dd29820a78e40d1c9aad2eb/src/Response.php#L92) as `page.url` which makes Inertia change the url like you described.

Instead of using Inertia, you could use a direct `fetch` or `Axios`. 

There's also an open [pull request](https://github.com/inertiajs/inertia-laravel/pull/206) which mitigates the issue by allowing you to specify the `page.url`.

**Reply by darklord1807** [reply]
Thanks a lot for this. I used the link and was able to make it work. It really helps.
